auth/auth-server/src/telemetry/helpers.rs

```diff
@@ -0,0 +1,154 @@
+/// the given mint and amount.
+/// The asset is the token ticker, if it is found, otherwise
+/// the token's address.
+/// The amount is the decimal amount of the transfer, going through
+/// lossy f64 conversion via the associated number of decimals
+fn get_asset_and_volume(mint: &str, amount: u128) -> (String, f64) {
+    let token = Token::from_addr(mint);
+    let asset = token.get_ticker().unwrap_or(mint.to_string());
+    let volume = token.convert_to_decimal(amount);
+
+    (asset, volume)
+}
+
+/// Calculates the quote per base price from a match bundle
+/// Returns the price as an f64 decimal adjusted value
+fn calculate_quote_per_base_price(match_bundle: &AtomicMatchApiBundle) -> f64 {
+    let (quote, base) = match match_bundle.match_result.direction {
+        OrderSide::Buy => (&match_bundle.send, &match_bundle.receive),
+        OrderSide::Sell => (&match_bundle.receive, &match_bundle.send),
+    };
+
+    Token::from_addr(&quote.mint).convert_to_decimal(quote.amount)
+        / Token::from_addr(&base.mint).convert_to_decimal(base.amount)
+}
+
+/// Converts a decimal amount to token native units, accounting for the token's
+/// decimals. This is the inverse operation of convert_to_decimal.
+fn convert_from_decimal(token: &Token, decimal_amount: f64) -> u128 {
+    let decimals = token.get_decimals().unwrap_or_default();
+    (decimal_amount * (10u128.pow(decimals as u32) as f64)) as u128
+}
+
+/// Record a volume metric with the given extra tags
+fn record_volume_with_tags(
+    mint: &str,
+    amount: u128,
+    volume_metric_name: &'static str,
+    extra_labels: &[(String, String)],
+) {
+    let (asset, volume) = get_asset_and_volume(mint, amount);
+    let mut labels = vec![(ASSET_METRIC_TAG.to_string(), asset)];
+    let extra_labels = extra_labels.iter().map(|(k, v)| (k.clone(), v.clone()));
+    labels.extend(extra_labels);
+
+    // We use a gauge metric here to be able to capture a float value
+    // for the volume
+    metrics::gauge!(volume_metric_name, labels.as_slice()).set(volume);
+}
+
+/// Records metrics for the incoming external match request
+fn record_external_match_request_metrics(
+    req: &ExternalMatchRequest,
+    price: f64,
+    labels: &[(String, String)],
+) -> Result<(), AuthServerError> {
+    // Record external order volume
+    let base_mint = biguint_to_hex_addr(&req.external_order.base_mint);
+    let quote_mint = biguint_to_hex_addr(&req.external_order.quote_mint);
+
+    // Calculate amount in base
+    let timestamped_price = TimestampedPrice::new(price);
+    let fixed_point_price = timestamped_price.as_fixed_point();
+    let order = req.external_order.to_order_with_price(fixed_point_price);
+
+    // Calculate amount in quote
+    let (_, volume) = get_asset_and_volume(&base_mint, order.amount);
+    let quote_token = Token::from_addr(&quote_mint);
+    let quote_amount = convert_from_decimal(&quote_token, volume * fixed_point_price.to_f64());
```

Comment:
no need to convert back the `fixed_point_price` price to `f64`, just use `price`.